Implemented ActorBBM in ActorInspector

Domily11 opened the pull request.

Domily11 committed:
modified actor model in joint.extentions

Added actortype parameter in joint.shapes.basic.Actor and placed a function to assign associated lines. Wouldn't show up without updating actorinspector

Domily11 committed:
Merge pull request #392 from amgrubb/mack-modifyactormodel

modified actor model in joint.extentions

Domily11 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Domily11 committed:
Implemented Backbone for ActorInspector

Actor Inspector is now as a backbone structure

Domily11 committed:
Fixed small bug of changeLine

Actor now has the associated line show up when you change the types

Domily11 committed:
Put placeholder in Actor input box

Domily11 committed:
Took off comments

Domily11 committed:
Resolved some comments

Took out the console.logs and extra lines and unnecessary codes

Domily11 committed:
Took out extra line

Domily11 committed:
Template fixed

Domily11 committed:
Temporary fix for the name bug

The actor's name now appears on the text box every time you visit the actor inspector

Manarnz committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Manarnz committed:
Fixed the inspector view problem

I connected the code with the changes Ali made and now it is working.

Domily11 committed:
Fix bug #408

Manarnz committed:
Added comments to the functions

To make reading the code easier, I added comments to the functions describing what each one does.

Manarnz committed:
Merge branch 'mack-actorinspector' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Manarnz committed:
Revert "Added comments to the functions"

This reverts commit a4e8aff46b84659a57a562b87d4a902df466a989.

Manarnz committed:
Added Comments To Functions

Manarnz committed:
Delete ConfigInspector.js from Commit

Manarnz committed:
Implemented small comments

Manarnz committed:
Merge branch 'mack-actorinspector' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Manarnz committed:
Revert "Delete ConfigInspector.js from Commit"

This reverts commit ff3b24e828139df8e674fc6ce56599f95c171418.

Manarnz committed:
Changed the way the actorType placeholder updates

Before, the placeholder is updated in the render function in JS, now it is updated in the template itself in the script.

Manarnz committed:
implement Ali's *final* comments

leshghi committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Domily11 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Domily11 committed:
Implemented ActorBBM in ActorInspector

Domily11 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -812,7 +797,10 @@ graph.on("add", function(cell) {
 
 	} else if (cell instanceof joint.shapes.basic.Actor) {
-		createActor(cell);
+        //TODO: right now there are two parameters the actor model in the joint.extensions file that hold the same information (attrs.name & actorName), find a way for actor inspector to be able to access attrs.name in the template script so that actorName is not needed
```
Remove comment

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -812,7 +797,10 @@ graph.on("add", function(cell) {
 	} else if (cell instanceof joint.shapes.basic.Actor) {
-		createActor(cell);
+        //TODO: right now there are two parameters the actor model in the joint.extensions file that hold the same information (attrs.name & actorName), find a way for actor inspector to be able to access attrs.name in the template script so that actorName is not needed
+		var name = cell.attr('.name/text') + "_" + Actor.numOfCreatedInstances;
```
Find a new way to create a name that doesnt use Actor.numOfCreatedInstances

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -4,19 +4,16 @@ var ENTER_KEY = 13;
 var ActorInspector = Backbone.View.extend({
         model: joint.shapes.basic.Actor,
         
-        initialize: function() {
```
Bring back initialize and use it to set this.actor = this.model.get('actor') so that you can access it easily throughout the file

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -34,7 +31,8 @@ var ActorInspector = Backbone.View.extend({
             // If the clicked node is an actor, render the actor inspector
-            this.$el.html(_.template($(this.template).html())(this.model.toJSON()));
+            this.$el.html(_.template($(this.template).html())(this.model.get('actor').toJSON()));
+            console.log(this.model.get('actor'))
```
remove console log

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -65,13 +63,10 @@ var ActorInspector = Backbone.View.extend({
-        changeLine: function() {
-            var actorType = this.model.get('actorType');
+            var actorType = $('#actor-type-ID').val();
+            //TODO use the following code and find a method to directly listen to the model instead of taking the information from the interface, 
```
Remove these comments and make an Issue for this instead

leshghi reviewed (changes requested):
Looks great small comments just to make the code cleaner but otherwise wonderful!

Domily11 committed:
Temporary easy solution for automatic naming actor

Domily11 committed:
Pulled from backbone

Domily11 committed:
Implemented the Embedding of Intentions in Actor (#442)

* Added structural changes for implementation of intentionBBM (#440)

* Modified embedding function

* BackBone Refactor for LinkInspector (#409)

* First Step of BackBone Refactor for LinkInspector

* Change the setSelectValue to get attributes directly from model

Also, used it in initialize

* Able to Call LinkInspector again

* changed code in render to single quotation marks

* Fixed problem where weren't able to choose evolving relationships

* Added the necessary code for clearInspector

* Attempt to switch to switchMode

* Fixed bugs from when I switched to switchMode

* Fixed dropdown menu problem for evolving relations

The "end" is not greyed out however

* Half Fixed the Dropdown Bug

Previously when we clicked switched between evolving and constant relationships the "begin" in the evolving relationship will be unclickable when you switch back to it the second time. Now it is but only if you did click update or choose anything at first in the begin dropdown

* Commit to save work

* Revert "Commit to save work"

This reverts commit 97c9bf9c4baee5e8ec78bd3bfe34bac678fbfc9b.

* Started scripting the template for evolving functions

* Fixed template and bug of dropdown menu

* Mack testtemplate (#429)

* Create BackboneModels.js

Create file for backbone model conversion

* Temporarily removing Load Sample Model button until backbone refactor is finished (#410)

* Add BackboneModels.js to index.html

* Moved onclick event to onFunctions.js (#412)

* Moved onclick event to onFunctions.js, copy paste functions work

* Moved keyboard functions and deleted old file

* removed keyboardFunctions.js from index.html

* Cleanup displayAnalysis (#422)

* Added backbone model representations for UserEvaluation and Constraint

* Deleted commented lines

* Removed dead functions and commented out lines

* Deleted more functions

* Removed extra file

* Deleted refreshAnalysisUI() switchResults() and switchSelectedShadingResult(). Made issue for refreshAnalysisUI

* Added some TODOs

Co-authored-by: leshghi <43756305+leshghi@users.noreply.github.com>

* Created FuncSegment and RepFuncSegment backbone models in BackboneModels.js (#421)

* made FuncSegment and RepFuncSegment backbone models

* Changed parameters in initialize function to single option parameter

* Updated variable names

* Update RepFuncSegment model name

* Update BackboneModels.js

Co-authored-by: angelac2023 <achu91@smith.edu>
Co-authored-by: A. M. Grubb <amgrubb@smith.edu>

* Added backbone model representations for UserEvaluation and Constraint (#420)

* Added backbone model representations for UserEvaluation and Constraint

* Deleted commented lines

* Made name changes and defaults

* added this.type

* Changed destRefTP name.

* Update BackboneModels.js

* removed defaults

Co-authored-by: A. M. Grubb <amgrubb@smith.edu>

* Tried another way of going with the template

* Commented thought process

* Updated more comments and took out unnecessary codes

* Added comment

Co-authored-by: leshghi <43756305+leshghi@users.noreply.github.com>
Co-authored-by: YesugenB <59470488+YesugenB@users.noreply.github.com>
Co-authored-by: i-m-foster <72589318+i-m-foster@users.noreply.github.com>
Co-authored-by: angelac2023 <achu91@smith.edu>
Co-authored-by: A. M. Grubb <amgrubb@smith.edu>

* Deleted extra lines

* Combining LinkBBM with LinkInspector (#432)

* Getting started

This is just us testing, nothing serious.

* Fixed issues in previous commit

We changed the placement of selected from LinkBBM to the model, as it is concerned with the interface and not backbone.
** In each function, we defiende link, is that okay?

* Fixed linkType and type problem

Another bug came up tho, that when you are switching between different link types it reassignes all the "linkType" in all the links

* Took out actorInspector.clear()

Co-authored-by: Domily11 <47505178+Domily11@users.noreply.github.com>

* Fixed the actorType issue and aded the error message

Previously, when the target is initally connected, it is not updated. To solve this, we created the updateTarget. We also added a feature, such that if the source is actor and the target is not, an error message would show up instead of the actor template.

* implemented Ali's comments

* deleted the app.js file

* deleted index.html

* Remove unwanted files from PR, fix smaller comments that were overlooked

* Some clean up for link inspector (#444)

* first round of cleanup

* More cleanup to LinkInspector.js

* Added structural changes for implementation of intentionBBM (#440)

* Reorg and comments

* Fix small bugs

* Remove repeat error since it does not accurately potray when link data is saved

Co-authored-by: i-m-foster <72589318+i-m-foster@users.noreply.github.com>

* Last fixes post merge

Co-authored-by: Manarnz <83615524+Manarnz@users.noreply.github.com>
Co-authored-by: leshghi <43756305+leshghi@users.noreply.github.com>
Co-authored-by: YesugenB <59470488+YesugenB@users.noreply.github.com>
Co-authored-by: i-m-foster <72589318+i-m-foster@users.noreply.github.com>
Co-authored-by: angelac2023 <achu91@smith.edu>
Co-authored-by: A. M. Grubb <amgrubb@smith.edu>

* Fixed issue with trashcan not showing up for RelativeIntentionView

* Pull from actorinspector

Co-authored-by: i-m-foster <72589318+i-m-foster@users.noreply.github.com>
Co-authored-by: Manarnz <83615524+Manarnz@users.noreply.github.com>
Co-authored-by: leshghi <43756305+leshghi@users.noreply.github.com>
Co-authored-by: YesugenB <59470488+YesugenB@users.noreply.github.com>
Co-authored-by: angelac2023 <achu91@smith.edu>
Co-authored-by: A. M. Grubb <amgrubb@smith.edu>

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -877,16 +866,24 @@ paper.on({
                         if (cell.get('parent')) {
                             graph.getCell(cell.get('parent')).unembed(cell);
-
+                            /** 
```
Can delete this

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -877,16 +866,24 @@ paper.on({
                             }
+                            */
                         }
                         // embed element in new actor
```
Capitalize the first letter of comments // Like this

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -65,13 +64,13 @@ var ActorInspector = Backbone.View.extend({
+            var actorType = $('#actor-type-ID').val();
+
+            //TODO use the following code and find a method to directly listen to the model instead of taking the information from the interface, 
+            //current problem is that actorType is within the actor parameter (actorBBM) but is dealing with attributes from the basic.Actor model
```
Still need to remove these comments

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -65,13 +64,13 @@ var ActorInspector = Backbone.View.extend({
+
+            //TODO use the following code and find a method to directly listen to the model instead of taking the information from the interface, 
+            //current problem is that actorType is within the actor parameter (actorBBM) but is dealing with attributes from the basic.Actor model
+            this.model.get('actor').set('type', actorType);
```
You already do this more efficiently in line 73
```suggestion
```

leshghi commented on leaf-ui/rappid-extensions/joint.extensions.js:
```diff
@@ -267,11 +268,9 @@ joint.shapes.basic.Actor = joint.shapes.basic.Generic.extend({
             }
         },
         //TODO replace with new ActorBBM once it is in codebase
```
Delete TODO

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -780,7 +766,10 @@ graph.on("add", function(cell) {
 
 	} else if (cell instanceof joint.shapes.basic.Actor) {
-		createActor(cell);
+        createdInstance = createdInstance.filter(view => view.model instanceof joint.shapes.basic.Actor);
```
Looks awesome. Add a comment for what this is doing, and make an issue for the potential duplicate names (if you make Actor_0, Actor_1, and then delete Actor_0 and make a new actor then you will have two Actor_1s for example)

leshghi reviewed (changes requested):
Awesome! Just respond to comments then looks good

Domily11 committed:
Responded to Comment

leshghi reviewed (approved):
This looks great - thank you!
